Utils/github_workflow_scripts/utils.py

```diff
@@ -92,25 +92,53 @@ class Checkout:  # pragma: no cover
     previously current branch.
     """
 
-    def __init__(self, repo: Repo, branch_to_checkout: str):
+    def __init__(self, repo: Repo, branch_to_checkout: str, fork_owner: Optional[str] = None):
         """Initializes instance attributes.
         Arguments:
             repo: git repo object
             branch_to_checkout: The branch or commit hash to check out.
+            fork_owner (str): The owner of the forked repository.
+                Leave it as None if the branch is in the same repository.
         """
         self.repo = repo
-        self.repo.remote().fetch(branch_to_checkout)
-        self._branch_to_checkout = branch_to_checkout
+
+        if fork_owner:
+            forked_remote_name = f'{fork_owner}_{branch_to_checkout}_remote'
```

Comment:
what is this name? just a name we use internally in the code or it is based on something related to contrib PRs?

Reply:
its the name of that remote that will be created in the workflow to the given repository of the forked contributor branch.

You can choose any name you want, in our case I thought its best to include the `fork_owner` + `branch_to_checkout` and maybe ill add also the name of the repo that we checkout to it.